custom/icds/repeaters/generators/phi.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+from django.core.serializers.json import DjangoJSONEncoder
+
+from corehq import toggles
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.repeater_generators import (
+    CaseRepeaterJsonPayloadGenerator,
+)
+
+
+class BasePayloadGenerator(CaseRepeaterJsonPayloadGenerator):
+    @staticmethod
+    def enabled_for_domain(domain):
+        return toggles.PHI_CAS_INTEGRATION.enabled(domain)
+
+
+class SearchByParamsPayloadGenerator(BasePayloadGenerator):
+    @staticmethod
+    def _gender(gender):
+        if gender:
+            if gender == 'male':
+                return 'M'
+            elif gender == 'female':
+                return 'F'
+        return ""
+
+    def get_payload(self, repeat_record, case):
+        data = {
+            "beneficaryname": case.name,
+            "fathername": case.get_case_property('fathers_name') or "",
+            "husbandname": case.get_case_property('husbands_name') or "",
+            "mothername": case.get_case_property('mothers_name') or "",
+            "gender": self._gender(case.get_case_property('gender')),
```

Comment:
Or 
```javascript
    "gender": CasePropertyMap(
        case_property="gender",
        value_map={
            "male": "M",
            "female": "F"
        }
    ).get_value(case_trigger_info)
```

Reply:
yayee, thanks

Reply:
building that `case_trigger_info` just to pass to this to fetch a case property looks a heavy lift for something simple, do you think this wont a good use case for using `CasePropertyMap`?

Reply:
Yeah, you'd really want the Repeater to create a CaseTriggerInfo instance so you could use it for all values on all API calls. Using it just for one value on one call doesn't sound like it would simplify the code; it would make it more complex.

If this is a one-off, possible temporary, integration, where you are hard-coding not just values, but also case property names, then a more generic solution might be a case of YAGNI.

But if this turns out not to be temporary, or if we might want to use it for more than one app where case property names might be different or hard-coded constants are a bad idea, then using `ValueSource` subclasses will give you the flexibility you need.